### ISSUE TITLE
reg-free COM on windows 10 pro

### ISSUE THREAD

**Issue by jmorris-spm** [VFPX/ParallelFox#8]
I saw the presentation at Southwest Fox last month, and downloaded the session files of Version 2.0.

I have been doing testing on Windows 7 using the "reg-free COM" option, and it works fine on Win 7.  Very impressive results. 

I tested on two different machines, and initially, both failed with registration errors the first time I tried, but on the second attempt they both worked, and worked fine afterwards.

Today I tried testing it on a brand new installation of Win 10 Pro, and it doesn't want to work.

Both Foxpro and Windows throw up "not registered" errors. I tried several times hoping the issue would correct itself as it did on Win 7, even using "Run as Administrator", but no luck.

Is there something additional I need to do on Win 10?

Thanks.
James Morris

**Comment by JoelLeach** [comment]
Hi James,

There should not be anything special required for Windows 10.  Without knowing more about the environment on those machines, here are a few possibilities off the top of my head:

- If you are using debug mode, the full VFP 9 needs to be installed on the machine.
- You may need to install the VFP 9 runtime files (or copy them into the same folder as ParallelFox.exe).  
- Parallel.SetRegFreeCOM() only applies to out-of-process workers  If you are using in-process workers with Parallel.SetMultiThreaded(.T.) then you either need to register ParallelFoxMT.dll or build a manifest into your EXE to use reg-free COM.  There is a link in the whitepaper to Rick Strahl's instructions for this.
- Parallel.SetRegFreeCOM() needs to be able to find ParallelFox.exe.  If it's not in your current Fox path, you can pass the location of the file as the second parameter.  If not found, it should produce a "File does not exist" error.

If none of that helps, please provide more detailed info on the error.

**Comment by jmorris-spm** [comment]
Hi Joel,

Thanks for the quick help. I copied the runtimes into the ParallelFox folder, and problem solved.

I placed the entire ParallelFox folder inside of my main folder where my executable is, with all of the runtimes in the main folder. Although I added the ParallelFox folder to my program's Fox path, it never occurred to me until I saw your answer that ParallelFox.exe inside that folder would also need to see the runtimes.

Thanks again.
James Morris
